services/wallet/async/scheduler.go

```diff
@@ -156,13 +166,15 @@ func (s *Scheduler) runTask(tc *taskContext, taskFn taskFunction, resFn func(int
 func (s *Scheduler) finishedTask(finishedRes interface{}, doneTask *taskContext, finishedResFn resultFunction, finishedErr error) {
 	s.queueMutex.Lock()
 
-	// We always have a running task
 	current := s.queue.Oldest()
 	// Delete current task if not overwritten
 	if s.doNotDeleteCurrentTask {
 		s.doNotDeleteCurrentTask = false
 	} else {
-		s.queue.Delete(current.Value.taskType)
+		// current maybe nil if Stop() is called
+		if current != nil {
```

Comment:
This solution looks simple and precise 👍 
But I wonder if we should just wait for all tasks to finish inside `Stop()`, after `s.cancelFn()` 🤔